fernet-aws-secrets-manager-rotator/src/main/java/com/macasaet/fernet/aws/secretsmanager/rotation/MultiFernetKeyRotator.java

```diff
@@ -0,0 +1,126 @@
+     * @param kms a KMS client for seeding the random number generator
+     * @param random an entropy source
+     */
+    protected MultiFernetKeyRotator(final SecretsManager secretsManager, final AWSKMS kms, final SecureRandom random) {
+        super(secretsManager, kms, random);
+        final String maxActiveKeysString = System.getenv("MAX_ACTIVE_KEYS");
+        if (maxActiveKeysString != null && !"".equals(maxActiveKeysString)) {
+            setMaxActiveKeys(Integer.parseInt(maxActiveKeysString));
+        }
+    }
+
+    public MultiFernetKeyRotator() {
+        this(new SecretsManager(AWSSecretsManagerClientBuilder.defaultClient()), AWSKMSClientBuilder.defaultClient(),
+                new SecureRandom());
+    }
+
+    protected void createSecret(final String secretId, final String clientRequestToken) {
+        final ByteBuffer currentSecret = getSecretsManager().getSecretStage(secretId, CURRENT);
+        if (currentSecret.remaining() % 32 != 0) {
+            throw new IllegalStateException("There must be a multiple of 32 bytes.");
+        }
+        final int numKeys = currentSecret.remaining() / 32;
+        List<Key> keys = new ArrayList<>(numKeys + 1);
+        while (currentSecret.hasRemaining()) {
+            final byte[] signingKey = new byte[16];
+            currentSecret.get(signingKey);
+            final byte[] encryptionKey = new byte[16];
+            currentSecret.get(encryptionKey);
+            final Key key = new Key(signingKey, encryptionKey);
+            keys.add(key);
+        }
+        final Key keyToStage = Key.generateKey(getRandom());
+        keys.add(0, keyToStage);
+        final int desiredSize = getMaxActiveKeys() + 1; // max active keys + one pending
+        if (keys.size() > desiredSize) {
+            keys = keys.subList(0, desiredSize);
+        }
+
+        getSecretsManager().putSecretValue(secretId, clientRequestToken, keys, PENDING);
+        getLogger().info("createSecret: Successfully put secret for ARN {} and version {}.", secretId,
+                clientRequestToken);
+    }
+
+    protected void testSecret(final String secretId, final String clientRequestToken) { 
+        final ByteBuffer currentSecret = getSecretsManager().getSecretVersion(secretId,
+                clientRequestToken);
+        if (currentSecret.remaining() % 32 != 0) {
+            throw new IllegalStateException("There must be a multiple of 32 bytes.");
+        }
+        // first key will become the staged key
+        final byte[] signingKey = new byte[16];
+        currentSecret.get(signingKey);
+        final byte[] encryptionKey = new byte[16];
+        currentSecret.get(encryptionKey);
+        final Key candidateStagedKey = new Key(signingKey, encryptionKey);
+        final Token token = Token.generate(getRandom(), candidateStagedKey, "");
+        if (!token.isValidSignature(candidateStagedKey)) {
+            throw new IllegalStateException("Staged key cannot generate a valid token.");
+        }
```

Comment:
The `Key` already performs validation in the constructor, so there is no need to validate that it can create and decrypt a token.